### PR TITLE
Add testing for sync_fd

### PR DESCRIPTION
**Pull request by lakshmih**
Modify the external semaphore extension test to use SYNC_FD, if available on the device. 
Also, made tests compatible with blocking semaphore spec updates (https://github.com/KhronosGroup/OpenCL-Docs/pull/882)

This PR incorporates the functionality from #1608 and supersedes it.

### CODE REVIEW

**Comment by JohnKesapidesARM** [comment]
I've been asked by @kpet to have a look on behalf of ARM. I'll try and get this reviewed by weeks end.

File: test_conformance/common/vulkan_wrapper/vulkan_utility.cpp [hunk 1]
```diff
@@ -248,6 +248,9 @@ getSupportedVulkanExternalSemaphoreHandleTypeList()
     }
     externalSemaphoreHandleTypeList.push_back(
         VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_OPAQUE_WIN32_KMT);
+#elif defined(__ANDROID__)
+    externalSemaphoreHandleTypeList.push_back(
+        VULKAN_EXTERNAL_SEMAPHORE_HANDLE_TYPE_SYNC_FD);
```
Comment (paulfradgley) [inline test_conformance/common/vulkan_wrapper/vulkan_utility.cpp:253]: SYNC_FD only on Android platforms?

**Comment by paulfradgley** [comment]
With SYNC_FD enabled on Linux platforms, the tests ran through and worked properly for me.

Is it the plan eventually to loop over all supported export handle types? Currently the tests only run with the first export handle_type in the vector.

**Comment by joshqti** [comment]
> With SYNC_FD enabled on Linux platforms, the tests ran through and worked properly for me.
> 
> Is it the plan eventually to loop over all supported export handle types? Currently the tests only run with the first export handle_type in the vector.

I agree, the current structure of the test is insufficient to test all handle types. I have a local change that generalized the test to iterate over all handle types.  It was originally as a part of https://github.com/KhronosGroup/OpenCL-CTS/pull/1676, however the PR was too large, and I removed all unnecessary refactoring and improvements.  This PR is the minimal change to test SYNC_FD.  Once this PR and 1676 merge, I can merge the changes to iterate over all handle types.  I was finding it difficult to push it as an independent PR due to massive merge conflicts that would occur.

File: test_conformance/common/vulkan_wrapper/vulkan_utility.cpp (hunk 1 above)
Comment (joshqti) [reply test_conformance/common/vulkan_wrapper/vulkan_utility.cpp:253]: We can add Linux

**Comment by lakshmih** [comment]
We will push a follow-on PR that cleans up the test structure once #1676 and this one merges. That will add support for iterating over all handle types.

**Review by bashbaug — APPROVED** [review]
Merging as discussed in the August 1st teleconference.

**Pull request merged**